### PR TITLE
[Doc] Release notes 2022.2

### PR DESCRIPTION
**Pull request by viniarck**
This is an initial version for the 2022.2 release notes. When consolidating this release notes from the core projects and NApps that are being released, I took a slightly different approach, I primarily focused on the goals of the version considering a more general audience for this type of document and then the more specific technical or implementation details that a network operator or developer might need to know more about is available on the CHANGELOG.rst of each NApp or README.rst. 

Team, let me know if you're OK with how the information was primarily structured, feel free to send commit suggestions or open new threads as needed. cc'ing @jab1982 @italovalcy @ajoaoff @rmotitsuki @Ktmi  @gretelliz 

If you don't have `jekyll` installed yet, you can also get an equivalent preview from this document directly from GitHub on [this link](https://github.com/kytos-ng/kytos-ng.github.io/blob/doc/release_notes_2022.2/releases/2022.2/2022.2.md)

There are some intentional `TODOs` in the document that'll open new threads in a moment.

### CODE REVIEW

File: releases/2022.2/2022.2.md [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  Figure 1 - Kytos NApps high level integration with MongoDB
+</div>
+
+On kytos-ng documentation, there are more technical information about the current [integration architecture](https://kytos-ng.github.io/napps/mongodb.html) with MongoDB and [general notes](https://kytos-ng.github.io/napps/mongodb_notes.html) for production deployment. On kytos [readme](https://github.com/kytos-ng/kytos#how-to-use-with-mongodb) page, there are more information about how to run and use MongoDB with Kytos.
+
+`flow_manager`, `mef_eline`, and `topology` have migrated from `storehouse` to MongoDB for higher performance, production and maintainability reasons. Also, since this new database infrastructure is officially supported, `storehouse` has been deprecated. If you were using any of these NApps in your network infrastructure, a migration script is also available in the `scripts/` folder of each NApp repository to migrate the stored data gracefully to leverage the new enhanced implementation and supported infrastructure.
+
+### 2. `mef_eline` EVCs convergence optimizations and QinQ enhancements
+
+TODO: Ask Italo to add and summarize the EVCs convergence gains.
```
Comment (viniarck) [inline releases/2022.2/2022.2.md:47]: @italovalcy, if you could please help out with this TODO that'd be awesome.

File: releases/2022.2/2022.2.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+`mef_eline` has reintroduced setting the customer VLAN on the ingress when pushing the QinQ encapsulation since this action is correctly supported by OvS and NoviFlow switches. TODO: if it is worth mentioning any potential unsupported case that network operators should be aware of or any other rewording about this paragraph.
```
Comment (viniarck) [inline releases/2022.2/2022.2.md:49]: @ajoaoff, if you could review this paragraph or any other information related to the TODO that you believe it's worth highlighting that'd be great

File: releases/2022.2/2022.2.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+
+### 6. Interface loop detection
+
+`of_lldp` has also been augmented to detect if an interface of a switch is looped. [Interface loop detection](https://github.com/kytos-ng/kytos/blob/master/docs/blueprints/EP025.rst) is automatically enabled once LLDP since it is been designed to detect unexpected or accidental loops that can happen. This feature has configuration options to configure configure which action should be applied once a loop is detected, by default, it generates a log warning message, but it can also disable an interface. Also, loops can be configured to be ignored.
+
+### 7. New Web UI components
+
+The following Web UI functionalities were added on `topology`:
```
Comment (viniarck) [inline releases/2022.2/2022.2.md:78]: @rmotitsuki if you could also double check the UI features or the presented information that'd be great. Let me know if you'd like to highlight any other major objective or point.

**Review by gretelliz — APPROVED** [review]

**Comment by viniarck** [comment]
I've pushed this new commit based on Jeronimo's feedback https://github.com/kytos-ng/kytos-ng.github.io/pull/34/commits/09190fe65a4a8172eef7a172fbfc0ffce5f259b3

File: releases/2022.2/2022.2.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+- Have a better infrastructure tools for application performance monitoring (APM) and logging.
+- New major features such as link liveness detection, interface loop detection, and new Web UI components.
+- Increase unit test coverage to at least 85% on `coloring`, `sdntrace`, and `sdntrace_cp`.
+- Leverage queue loggers to minimize latency.
+
+In order to achieve the objectives of this version, the following enhancements were refined and delivered:
+
+### 1. `mef_eline` EVCs convergence optimizations
+
+TODO: Ask Italo to add and summarize the EVCs convergence gains.
```
Comment (italovalcy) [inline releases/2022.2/2022.2.md:32]:
```suggestion
> Protection in Metro Ethernet Networks is a self-healing property of the network that allows it to continue to function with minimal or no impact on the network users upon disruption, outages, or degradation of links, facilities, or equipment. (MEF 2, Technical Specification)

Aligned with MEF Ethernet Service Protection technical specification, `mef_eline` has been enhanced with strategies to provide fast convergence for fully dynamic EVCs, as documented in  [EP029 blueprint](https://github.com/kytos-ng/kytos/blob/master/docs/blueprints/EP029.rst). In particular, we've introduced an improved convergence time when handling link failure events using maximum disjoint path algorithms and pre-installed failover paths.

`mef_eline` convergence performance is more than 100x faster in handling link failure events for hundreds of EVCs compared to the previous release (see complete performance evaluation and details in [PR #175](https://github.com/kytos-ng/mef_eline/pull/175). In addition, the overall performance achieved a sub-second convergence time (plus control plane latency) for handling a single link failure.
```

**Review by italovalcy — CHANGES_REQUESTED** [review]

File: releases/2022.2/2022.2.md (hunk 4 above)
Comment (viniarck) [reply releases/2022.2/2022.2.md:32]: These two paragraphs look great, very handy to also have a link of PR 175. Thanks @italovalcy, let me commit this.

File: releases/2022.2/2022.2.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+## Objectives
+
+This version had these major objectives:
+
+- Optimize `mef_eline` EVCs convergence to take place in a few seconds on average.
+- Integrate a production-grade recommended document-oriented database for Napps.
+- Enhance Kytos core to support event message prioritization and have higher performance when handling events in general.
+- Have a better infrastructure tools for application performance monitoring (APM) and logging.
+- New major features such as link liveness detection, interface loop detection, and new Web UI components.
+- Increase unit test coverage to at least 85% on `coloring`, `sdntrace`, and `sdntrace_cp`.
```
Comment (italovalcy) [inline releases/2022.2/2022.2.md:25]: Wouldn't it be on all Napps ? (then, to achieve the goal we had to work on coloring, sdntrace, and sdntrace_cp). I mean: now all Kytos Napps have more than 85% coverage, right? Maybe we can rephrase this to give this idea.

File: releases/2022.2/2022.2.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+<div align="center">
+  <a href="assets/maintenance_ui.png"><img src="assets/maintenance_ui.png"></a>
+</div>
+<div align="center">
+  Figure 4 - <code>maintenance</code> Web UI
+</div>
+
+### 9. Increase unit test coverage to 85% on `coloring`, `sdntrace` and `sdntrace_cp`
+
+`coloring`, `sdntrace`, and `sdntrace_cp` now have at least 85% of total unit test coverage, enhancing code quality and future code maintainability.
```
Comment (italovalcy) [inline releases/2022.2/2022.2.md:114]: here again, we can mention that all the Napps have coverage above 85%, and we can then mention those specific Napps. Wdyt?

**Review by italovalcy — APPROVED** [review]
Excellent release notes, @viniarck! Very well documented, written, and also very meaningful!

File: releases/2022.2/2022.2.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+`mef_eline` convergence performance is more than 100x faster in handling link failure events for hundreds of EVCs compared to the previous release (see complete performance evaluation and details in [PR #175](https://github.com/kytos-ng/mef_eline/pull/175)). In addition, the overall performance achieved a sub-second convergence time (plus control plane latency) for handling a single link failure.
+
+### 2. `mef_eline` QinQ enhancements
+
+`mef_eline` has reintroduced setting the customer VLAN on the ingress when pushing the QinQ encapsulation since this action is correctly supported by OvS and NoviFlow switches. TODO: if it is worth mentioning any potential unsupported case that network operators should be aware of or any other rewording about this paragraph.
```
Comment (ajoaoff) [inline releases/2022.2/2022.2.md:40]:
```suggestion
QinQ had been removed from `mef_eline` due to the lack of compatibility from Noviflow switches. The feature has been reintroduced by setting the customer VLAN on the ingress when pushing the QinQ encapsulation since this action is correctly supported by OvS and NoviFlow switches.
```
@viniarck @italovalcy

File: releases/2022.2/2022.2.md (hunk 5 above)
Comment (viniarck) [reply releases/2022.2/2022.2.md:25]: That's a good way to put it, @italovalcy. `flow_stats` is an exception though if we phrase it as all Kytos NApps, since we strategically decided not to try to cover it since it'll have major refactoring in the future. Maybe we can claim that in the next release, or maybe we can also come up with another adjective here, like all non-experimental NApps, and then we tag `flow_stats` with an `experimental` stability badge, let me know what you prefer to have here. 

I was also going to suggest to say all stable NApps, but some NApps haven't got the `stable` badge yet.

File: releases/2022.2/2022.2.md (hunk 7 above)
Comment (viniarck) [reply releases/2022.2/2022.2.md:40]: @ajoaoff appreciated this update, thanks

File: releases/2022.2/2022.2.md (hunk 5 above)
Comment (italovalcy) [reply releases/2022.2/2022.2.md:25]: Very interesting point, @vinicius. What do you think about leaving it as "all Kytos Napps (except flow_stats, which has a major refactoring coming soon)"?

Comment (viniarck) [reply releases/2022.2/2022.2.md:25]: Great idea @italovalcy, let's go for it.

Comment (viniarck) [reply releases/2022.2/2022.2.md]: @italovalcy done here https://github.com/kytos-ng/kytos-ng.github.io/pull/34/commits/4980a9a785cd3fab3ec99da0dd3f1a7d0597acbb

Comment (viniarck) [reply releases/2022.2/2022.2.md]: @italovalcy done here https://github.com/kytos-ng/kytos-ng.github.io/pull/34/commits/4980a9a785cd3fab3ec99da0dd3f1a7d0597acbb

File: releases/2022.2/2022.2.md (hunk 6 above)
Comment (viniarck) [reply releases/2022.2/2022.2.md]: @italovalcy done here https://github.com/kytos-ng/kytos-ng.github.io/pull/34/commits/4980a9a785cd3fab3ec99da0dd3f1a7d0597acbb

File: releases/2022.2/2022.2.md (hunk 3 above)
Comment (rmotitsuki) [reply releases/2022.2/2022.2.md:85]: @viniarck It looks like all the main goals are listed.
Can you try replacing the Kytos UI screenshots with a 4:3 or 16:9 aspect ratio? The second image is too stretched.

Comment (viniarck) [reply releases/2022.2/2022.2.md:85]: Right. Good idea @rmotitsuki, done here https://github.com/kytos-ng/kytos-ng.github.io/pull/34/commits/da2880b746725e698d8830a3e3904a65438cb3ca

Comment (viniarck) [reply releases/2022.2/2022.2.md:85]: I'll mark this as resolved. Thanks.

File: releases/2022.2/2022.2.md (hunk 5 above)
Comment (viniarck) [reply releases/2022.2/2022.2.md]: I'll mark this as resolved. Thanks.

**Comment by viniarck** [comment]
Thanks for the reviews, team. Lots of goodness and improvements.
